### DISCUSSION TITLE
how to insert snippet by click a webview button

### DISCUSSION THREAD

**Discussion by asevenm** [microsoft/vscode-discussions#405] [Extension Development QnA] [answered]
I'm developing a webview extention to manage snippets and click snippet item it can insert snippet to the selected textEditor. But I'm not familar with vscode api, I use fs.write instead. is there any method to achive the good user experience such as  input 'prefix' in textEditor?

**Comment by alefragnani** [comment, marked as answer]
Hi,

You should use `TextEditor.edit` API. A fairly simple example is available in the vscode-extensions-sample repo (https://github.com/microsoft/vscode-extension-samples/tree/main/document-editing-sample).

And, if you don't know how your webview should _to talk_ to it, maybe this more complete sample should fit your needs (https://github.com/microsoft/vscode-extension-samples/tree/main/webview-view-sample)

Hope this helps

**Reply by asevenm** [reply]
Thank you very much!  it definitely helps me.
